src/kvstore/wal/FileBasedWalIterator.cpp

```diff
@@ -16,9 +16,10 @@ FileBasedWalIterator::FileBasedWalIterator(
     std::shared_ptr<FileBasedWal> wal,
     LogID startId,
     LogID lastId)
-        : wal_(std::move(wal))
+        : wal_(wal)
         , currId_(startId) {
-    if (lastId >= 0) {
+    holder_ = std::make_unique<folly::RWSpinLock::ReadHolder>(wal_->rollbackLock_);
+    if (lastId >= 0 && lastId <= wal_->lastLogId()) {
```

Comment:
Why is the condition of "lastId <= wal _-> lastLogId() " added here?

Reply:
Cause we can't have bigger log id than last log id in wal.

Reply:
> Cause we can't have bigger log id than last log id in wal.

Why does this happen？

Reply:
> > Cause we can't have bigger log id than last log id in wal.
> 
> Why does this happen？

For example, leader want to send some latest log, and acquire a iterator, however the wal is rollbacked.